fabric-dimension-patches-v1/src/main/java/net/fabricmc/fabric/mixin/MixinDimensionRawIndexFix.java

```diff
@@ -0,0 +1,21 @@
+package net.fabricmc.fabric.mixin;
+
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.dimension.DimensionType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(DimensionType.class)
+public class MixinDimensionRawIndexFix
+{
+	@Inject(at = @At("HEAD"), method = "byRawId", cancellable = true)
+	private static void byRawId(final int id, final CallbackInfoReturnable<DimensionType> info) {
+		Registry.DIMENSION.forEach(dimension -> {
+			if(dimension.getRawId() == id) {
+				info.setReturnValue(dimension);
```

Comment:
This should probably use a regular for each expression (`for (DimensionType dimension : Registry.DIMENSION)`) and return early when something is found. It is a minor thing, but it would be slightly more efficient.